components/KillSwitchMonitor.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {useEffect} from 'react';
+import {Modal} from 'react-native';
+
+import * as Application from 'expo-application';
+import * as Updates from 'expo-updates';
+import _ from 'lodash';
+import PostHog, {PostHogProvider, useFeatureFlag, usePostHog} from 'posthog-react-native';
+
+import NoConnection from 'assets/illustrations/NoConnection.svg';
+import {Outcome} from 'components/content/Outcome';
+import {VStack} from 'components/core';
+import {useAppState} from 'hooks/useAppState';
+import {logger} from 'logger';
+
+type KillSwitchMonitorProps = {
+  children: React.ReactNode;
+};
+
+const tryReloadFeatureFlags = (posthog: PostHog) => {
+  logger.debug('fetching feature flags');
+  void posthog.reloadFeatureFlagsAsync();
+};
+
+// In release and preview mode, When returning to foreground, don't fetch feature flags more frequently than every 30 minutes
+// In development mode, refresh every time we return from foreground
+const FEATURE_FLAG_REFRESH_INTERVAL_MS = Updates.channel ? 30 * 60 * 1000 : 0;
```

Comment:
Is `Updates.channel` set on "vanilla" versions of the app - published once to the app stores and not sent to the user over OTA/EAS build?

Reply:
yes, channel is defined in `eas.json` for each build flavor. that value defines which channel the vanilla build would look at for its updates.